src/v/finjector/stress_fiber.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+
+#include <memory>
+
+struct stress_config {
```

Comment:
maybe it's also makes sense to add an optional list of shards that should be affected

Reply:
Filed an issue for some cleanup, but merged to unblock another PR
https://github.com/redpanda-data/redpanda/issues/12579